wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 2 - Post Card Navigation and Style Updates

Parent #15200

## Description

This PR

1. Adds navigation from the post cards (edit post navigation yet to be added as currently a mocked post is being used).
2. Add item decorator to manage "My Site" card and item margins (with exceptions: "Site Info Card" and "Quick Start Dynamic Card". "Site Info Card" requires a quick start icon focus point outside card margins and "Quick Start Dynamic Card" requires task type items to scroll to/ from the screen edges).
3. Fixes max-width for the "My Site" layout for the larger screens. 35a407f
4. Applies minor text style updates to "Domain Registration Card" and "Quick Start Dynamic Card". 

## Testing Instructions

PS: The post cards are currently displayed using mocked JSON response present under res/raw/mocked_posts_data.json. In order to test visibility for the cards, the mocked JSON will need to be updated appropriately as mentioned in the following tests. Let me know if anyone needs help with it. 🙂

See the [previous PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15563) to see instructions on how to enable each post card.

Prerequisite:

- Go to My Site -> Me -> App Settings -> Test Feature Configuration.
- Turn the MySiteDashboardPhase2FeatureConfig feature flag ON.
- Relaunch the app.

**Test Case 1: Post Card Navigation**

1. Create First/ Create Next Post Card
   - Click "Create a post" link.
   - Notice that the editor opens to create a new post.

2. Draft Post Card
   - Click "Go to drafts" link.
   - Notice that posts list screen opens with the `DRAFTS` tab selected.

3. Scheduled Post Card
   - Click "Go to scheduled posts" link.
   - Notice that posts list screen opens with the `SCHEDULED` tab selected.

4. Post Item
    - Click a post item in the "Draft" or "Scheduled" post card.
    - Since mocked posts data is used, this navigation is not working right now. Note that  [`TODO`](https://github.com/wordpress-mobile/WordPress-Android/pull/15606/commits/fd96b241514a57d8431e6da41bdda8f2fd577567#diff-2eaba53723de7ebd10ea23c2ac0022a6d91a53da055da5e0d04fddefe7a27d9dR344) is added to support it when actual posts are integrated into the flow.

### UI Related

**Test Case 2: My Site layout max width - Tablets**
- Load the "My Site" tab on a tablet. 
- Notice that the max-width constraint is applied to the layout.
   <img width="400" src="https://user-images.githubusercontent.com/1405144/143391850-ba1bb3f8-424e-43f9-8b04-9d7ae6930691.png"/>

PS: This got broken after adding swipe-to-refresh to the XML layout.

**Test Case 3: Register Domain Card**
- Using a site-switcher, switch to a paid site to see "Register Domain" card.
- Notice that the post card text styles and card styles are applied to it. 
      <img width="320" src="https://user-images.githubusercontent.com/1405144/143389071-bbacfcd4-82c8-4f7b-ab89-bdf0b7c6d679.jpg"/>

**Test Case 4: Quick Start dots**
- Fresh install the app or clear storage to enable the quick start flow for a site.
- Complete "Quick Start" steps to see that the "Quick Start" dots are not cut off on the cards (specifically on the Site Info card -> Site Icon) and items. 
    <img width="320" alt="quick start dot" src="https://user-images.githubusercontent.com/1405144/143391035-5dbe2d8e-96a4-46c5-8e69-a130c833ee90.png">

**Test Case 5: Card and Items Margins**
- Since the margins are now set using an item decoration, notice that they are set correctly on all the cards. 
- Note (in the [code](https://github.com/wordpress-mobile/WordPress-Android/pull/15606/commits/448171c6584044e2bedc2604e6a4578444069616#diff-27e9456f7875790ffe05af9be15bb32bfa73ce92ed25dfcb4f4f51697c4ed9fdR25-R26)) that 
    - Item decoration for "Quick Start Dynamic Card" is skipped as the quick start task type items in the dynamic card scroll to/ from screen edges and do not require any card margins.
    - Item decoration for the "Site Info Card" is skipped to allow the quick start dot to appear correctly outside the site icon layout bound.

## Merge Instructions
- Make sure that the design review is done.
- Remove `Needs Design Review` and `Not Ready for Merge` labels.
- Merge the PR.

## Regression Notes
1. Potential unintended areas of impact
See above scenarios.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
See above tests.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the navigation.

PR submission checklist:
